### PR TITLE
Sendinblue: allow using template's subject

### PR DESCRIPTION
**Pull request by tofran**
Adds the possibility for sending an email without providing the subject, and therefore using the one specified in the template.

Same as what was requested on #537, although this time it does not error, it provides an empty struct because of:
```elixir
defmodule Swoosh.Email do
  # (...)
  defstruct subject: "",
```
`lib/swoosh/email.ex` 

Related to #537

### CODE REVIEW

**Comment by princemaple** [comment]
This has nothing to do with #556 , subject has never been required.

**Comment by tofran** [comment]
Thank you @princemaple.
Yes you are right its not related to said issue.

I didn’t go this route because now you are preventing the usage of an empty subject.
Although thats not very desirable in a real life scenario, its the default swoosh behaviour on all other adapters.
From my side doesn’t make any difference tho

**Comment by princemaple** [comment]
If you don't set a template id, it'd still be empty subject, no?

File: lib/swoosh/adapters/sendinblue.ex [hunk 1]
```diff
@@ -119,6 +119,8 @@ defmodule Swoosh.Adapters.Sendinblue do
     Map.put(payload, "bcc", Enum.map(bcc, &prepare_recipient/1))
   end
 
+  defp prepare_subject(payload, %{subject: ""}), do: payload
```
Comment (tofran) [inline lib/swoosh/adapters/sendinblue.ex:122]: Maybe changing this to 
```elixir
defp prepare_subject(payload, %{subject: ""}), do: payload
```
would solve the inconstancy

**Comment by tofran** [comment]
I think we are not doing that check - but should
Thats a good test scenario

**Comment by princemaple** [comment]
I meant, does it make a difference, when you send `""` as subject to sendinblue, and when you not send a subject?

**Comment by tofran** [comment]
*Somehow my comment is now showing, if this is a dupe sorry please ignore.

Yes thats a good point, looks like they actually don't allow `"subject": "",`:

```json
{
  "code": "missing_parameter",
  "message": "subject is required"
}
```

All ok, case closed

**Comment by princemaple** [comment]
Thank you very much for the quick responses and the confirmation 🎉

**Pull request merged**